website/docs/d/datasource_google_service_account_id_token.html.markdown

```diff
@@ -0,0 +1,128 @@
+
+  There are further two variations on how this mode is used depending on how the source token is acquired.
+
+  1. If the source token is provided directly as a variable using the requestors token, then grant the requestor the `TokenCreator` role on the target service account.  In the following case, the identity in context with `gcloud` should have the TokenCreator role on `impersonated-account@project.iam.gserviceaccount.com`
+
+
+  Then run the terraform apply and specify the token directly
+
+     ```terraform apply --var static_access_token=`gcloud auth print-access-token` ```
+
+  ```yaml
+  provider google {}
+
+  variable "static_access_token" {
+    type = string
+  }
+
+  data "google_service_account_id_token" oidc {
+
+    // The following requires the owner of static_access_token to have
+    // ServiceAccountTokenCreator permissions on  
+    // impersonated-account@some_project.iam.gserviceaccount.com 
+    access_token = var.static_access_token
+    
+    // for any of the static or impersonated types provided
+    target_service_account = "impersonated-account@project.iam.gserviceaccount.com"
+    // delegates = []
+    include_email = true
+
+    target_audience = "https://foo.bar/"
+
+  }
+
+  output "oidc_token" {
+    value = data.google_service_account_id_token.oidc.id_token
+  }  
+  ```
+
+2. If the source token is provided by [google_service_account_access_token](https://www.terraform.io/docs/providers/google/d/datasource_google_service_account_access_token.html), then the target service account must have the `TokenCreator` role _on itself`.
+
+  >> Note: this variation should be extremely rare since you should be able to directly use the source system's service account JSON file directly.
+
+  ```yaml
+  provider google {}
+
+  data "google_service_account_access_token" "default" {
+  provider = google
+  target_service_account = "impersonated-account@project.iam.gserviceaccount.com"
+  delegates = []
+  scopes = ["userinfo-email", "cloud-platform"]
+  lifetime = "300s"
+  }
+
+  provider google {
+    alias  = "impersonated"
+    access_token = data.google_service_account_access_token.default.access_token
+  }
+
+  data "google_service_account_id_token" oidc {
+    provider = google.impersonated
+    
+    target_service_account = "impersonated-account@project.iam.gserviceaccount.com"
+    include_email = true
+
+    target_audience = "https://foo.bar/"
+  }
+
+  output "oidc_token" {
+    value = data.google_service_account_id_token.oidc.id_token
+  }
+  ```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `target_audience` (Required) - The audience claim value to to issued the `id_token` for.
+* `access_token` (Optional) - Raw access_token to use as the source credential.  Specifying this value will use IAMCredentials API even if any other variable is set.
```

Comment:
```suggestion
* `access_token` (Optional) - Raw access_token to use as the source credential.  Specifying this value will use the IAMCredentials API even if any other variable is set.
```

If this  is the case, do you want to make it impossible for people to set this along with other fields? If so, you can set `ConflictsWith` in the schema